### PR TITLE
authz: protect VPC Subnet endpoints

### PR DESCRIPTION
**Pull request by davepacheco**
This is a work-in-progress but I'm hopeful that it's pretty close:

- [x] test oxapi_demo
- [x] update failing tests
- [x] verify that my update to the "unauthorized" test correctly fails prior to my changes

This was a little trickier than some of the other ones for a few reasons:

- It's the first level of hierarchy below Projects that's not a leaf.  I made a kind of cheesy generalization to authz::ProjectChild to make this work.  I have some thoughts on making this more type-safe but I'd like to defer that for now because it'll affect many of the resources in that file.
- There's a saga using some of these functions so I had to update it to use the saga facility for serializing creds

@bnaecker I didn't realize until I got pretty far into this that this will likely conflict with #713.  I don't think the conflict will be too bad in either direction but it'll involve a bunch of manual fix-up.  I'll take another look at #713 before trying to land this.

### CODE REVIEW

File: nexus/tests/integration_tests/vpc_subnets.rs [hunk 1]
```diff
@@ -112,18 +127,24 @@ async fn test_vpc_subnets(cptestctx: &ControlPlaneTestContext) {
     assert_put_400(
         client,
         &subnet_url,
-        String::from("{ \"ipv6_block\": \"10.1.9.32/16\" }"),
+        &json!({ "ipv6_block": "10.1.9.32/16" }),
```
Comment (bnaecker) [inline nexus/tests/integration_tests/vpc_subnets.rs:130]: Robert and I were just discussing this in an RFD PR, but we'll ultimately only allow IP subnets that use the actual network address consistent with their mask, i.e., `"10.1.0.0/16"` in this case. Not a big deal now, just wanted to flag it.

Comment (rmustacc) [reply nexus/tests/integration_tests/vpc_subnets.rs:130]: If you have thoughts on that @davepacheco would definitely appreciate your feedback.

**Review by bnaecker — APPROVED** [review]
Looks great, thanks for this!

File: nexus/tests/integration_tests/vpc_subnets.rs (hunk 1 above)
Comment (davepacheco) [reply nexus/tests/integration_tests/vpc_subnets.rs:130]: That seems reasonable.  (I assume this is a bigger change than just changing this so I assume that'll be in another PR.)

**Pull request merged**